contracts/src/main/kotlin/com/contractsAndStates/states/GameBoardState.kt

```diff
@@ -133,82 +145,102 @@ data class PlacedHexTiles(val value: MutableList<HexTile>) {
                         val aboveRowLength = rowIndices[rowIndex - 1].count()
                         // If your row is shorter than above, you have 2 neighbors above
                         if (rowLength < aboveRowLength) {
-                            value[tileIndex].connect(TileSideIndex(5), value[tileIndex - aboveRowLength])
-                            value[tileIndex].connect(TileSideIndex(0), value[tileIndex - aboveRowLength + 1])
+                            value[columnIndex].connect(TileSideIndex(5), value[columnIndex - aboveRowLength])
+                            value[columnIndex].connect(TileSideIndex(0), value[columnIndex - aboveRowLength + 1])
                         }
                     }
                     // On each row, the last one has no neighbor to its right
                     if (index < rowLength - 1) {
-                        value[tileIndex].connect(TileSideIndex(1), value[tileIndex + 1])
+                        value[columnIndex].connect(TileSideIndex(1), value[columnIndex + 1])
                     }
                 }
             }
         }
 
+        /**
+         * This method is used in flows to product a new version of the gameboard with a record of the location of roads, identified by
+         * their specific linearID
+         *
+         * TODO: Add functionality to connect roadStates when new roads and proposed extending existing roads.
+         */
+        @Suspendable
+        fun buildRoad(
+                tileIndex: HexTileIndex,
+                sideIndex: TileSideIndex,
+                roadStateLinearId: UniqueIdentifier) = apply {
+            value[tileIndex.value].sidesBuilder
+                    .setRoadIdOn(sideIndex, roadStateLinearId)
+                    .getNeighborOn(sideIndex)?.also { neighborIndex ->
+                        value[neighborIndex.value].sidesBuilder
+                                .setRoadIdOn(sideIndex.opposite(), roadStateLinearId)
+                    }
+        }
+
         fun build() = PlacedHexTiles(ImmutableList(value.map { it.build() }).toMutableList())
     }
 }
 
 @CordaSerializable
-data class HexTile(val resourceType: HexTileType,
-                   val roleTrigger: Int,
-                   val robberPresent: Boolean,
-                   val hexTileIndex: HexTileIndex,
-                   val sides: HexTileNeighbors = HexTileNeighbors(),
-                   var roads: MutableList<UniqueIdentifier?> = MutableList(SIDE_COUNT) { null }) {
-
-    init {
-        require(roads.size == SIDE_COUNT) { "roads.size cannot be ${roads.size}" }
-    }
+data class HexTile(
+        val resourceType: HexTileType,
+        val roleTrigger: Int,
```

Comment:
@rogersanick 

* Should this be `rollTrigger`?
* I thought first that this is a number between `2` and `12`, but I have seen you set it to `0`? What is this?

Reply:
Hey @xavierlepretre - the rollTrigger is set to 0 where the hexTile is of a type that should never be triggered (e.g. 'Desert').

We could also change the type to be `RollTrigger?` but that might break the filtering logic in issue resources flow where the initiating player must sort for valid settlements.